packages/fxa-settings/src/pages/Authorization/container.tsx

```diff
@@ -0,0 +1,159 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { RouteComponentProps, useLocation, useNavigate } from '@reach/router';
+import {
+  Integration,
+  isDefault,
+  OAuthWebIntegration,
+  RelierAccount,
+  useAuthClient,
+  useSession,
+} from '../../models';
+import { cache } from '../../lib/cache';
+import { useCallback, useEffect, useState } from 'react';
+import { currentAccount } from '../../lib/cache';
+import { useFinishOAuthFlowHandler } from '../../lib/oauth/hooks';
+import OAuthDataError from '../../components/OAuthDataError';
+import { cachedSignIn, handleNavigation } from '../Signin/utils';
+import { AuthError, OAuthError } from '../../lib/oauth/oauth-errors';
+import { AuthUiErrors } from '../../lib/auth-errors/auth-errors';
+import { hardNavigate } from 'fxa-react/lib/utils';
+
+const convertToRelierAccount = (
+  account: ReturnType<typeof currentAccount>,
+  authClient: ReturnType<typeof useAuthClient>
+) => {
+  const relierAccount: RelierAccount = {
+    uid: account?.uid!,
+    email: account?.email!,
+    sessionToken: account?.sessionToken!,
+    verifyIdToken: authClient.verifyIdToken,
+    isDefault: () => isDefault(account || {}),
+  };
+  return relierAccount;
+};
+
+/**
+ * Unlike the other containers, this one does not attempt to render anything.
+ * It handles a 'prompt=none' authorization or hand off to the signin routes.
+ */
+const AuthorizationContainer = ({
+  integration,
+}: {
+  integration: Integration;
+} & RouteComponentProps) => {
+  const [oauthError, setOauthError] = useState<AuthError | OAuthError | null>(
+    null
+  );
+  const navigate = useNavigate();
+  const authClient = useAuthClient();
+  const location = useLocation() as ReturnType<typeof useLocation>;
+  const session = useSession();
+  const { finishOAuthFlowHandler, oAuthDataError } = useFinishOAuthFlowHandler(
+    authClient,
+    integration
+  );
+  if (oAuthDataError) {
+    setOauthError(oAuthDataError);
+  }
+
+  const promptNoneHandler = useCallback(async () => {
+    debugger;
+    const account = currentAccount();
+    const relierAccount = convertToRelierAccount(account, authClient);
+
+    try {
+      await (integration as OAuthWebIntegration).validatePromptNoneRequest(
```

Comment:
Instead of casting `integration as OAuthWebIntegration`, we can set the `integration` prop being passed in to `AuthorizationIntegration` and create a partial type for it. See what was done for the Index container page for example.

Reply:
`OAuthWebIntegration` is a subtype of `Integration` though.